pepper-apis/src/main/java/org/broadinstitute/ddp/util/RouteUtil.java

```diff
@@ -100,4 +104,49 @@ public static String parseStudyGuid(String path) {
         }
         return studyGuid;
     }
+
+    /**
+     * Get the activity instance, ensuring the study/user/instance exists and the instance is accessible. Otherwise, will set the
+     * appropriate route response.
+     *
+     * @param response        the route response
+     * @param handle          the database handle
+     * @param participantGuid the user guid
+     * @param studyGuid       the study guid
+     * @param instanceGuid    the activity instance guid
+     * @return the activity instance dto
+     */
+    public static ActivityInstanceDto findAccessibleInstanceOrHalt(Response response, Handle handle,
+                                                                   String participantGuid, String studyGuid, String instanceGuid) {
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyGuid);
+        if (studyDto == null) {
+            String msg = "Could not find study with guid " + participantGuid;
+            throw ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.STUDY_NOT_FOUND, msg));
+        }
+        User user = handle.attach(UserDao.class).findUserByGuid(participantGuid)
+                .orElseThrow(() -> {
+                    String msg = "Could not find user with guid " + participantGuid;
+                    return ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.USER_NOT_FOUND, msg));
+                });
+        ActivityInstanceDto instanceDto = handle.attach(JdbiActivityInstance.class)
+                .getByActivityInstanceGuid(instanceGuid)
+                .orElseThrow(() -> {
+                    String msg = "Could not find activity instance with guid " + instanceGuid;
+                    return ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.ACTIVITY_NOT_FOUND, msg));
+                });
+
+        if (instanceDto.getStudyId() != studyDto.getId() || instanceDto.getParticipantId() != user.getId()) {
+            LOG.warn("Activity instance {} does not belong to participant {} in study {}", instanceGuid, participantGuid, studyGuid);
+            String msg = "Could not find activity instance with guid " + instanceGuid;
+            throw ResponseUtil.haltError(response, 404, new ApiError(ErrorCodes.ACTIVITY_NOT_FOUND, msg));
+        } else if (instanceDto.isHidden()) {
+            String msg = "Activity instance " + instanceGuid + " is hidden";
+            throw ResponseUtil.haltError(response, 422, new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, msg));
```

Comment:
Is `422` okay? Should we use something else?

Reply:
resource is hidden and not supposed to be accessed . 403 should be good too.

Reply:
Would we be better off following API behaviors from examples such as Github- not coping to the instance existing at all and just returning a 404?

Reply:
I have been debating between 404, 422, etc. If folks are inclined, I can move to 404.

Reply:
Actually, having one less response type to deal with might be nice. Since clients can't do much in this case and it shouldn't happen that often, I'll move to 404 for now.